pinot-core/src/test/java/org/apache/pinot/queries/ForwardIndexHandlerReloadQueriesTest.java

```diff
@@ -554,23 +605,66 @@ public void testSelectWithAggregationQueriesWithReload()
+    assertEquals(brokerResponseNative.getNumSegmentsMatched(), 4L);
+    assertEquals(brokerResponseNative.getNumEntriesScannedPostFilter(), 400000);
+    assertEquals(brokerResponseNative.getNumEntriesScannedInFilter(), 0L);
+    assertNotNull(brokerResponseNative.getProcessingExceptions());
+    assertEquals(brokerResponseNative.getProcessingExceptions().size(), 0);
+    assertEquals(resultTable.getDataSchema(), new DataSchema(new String[]{"max(arraylength(column6))"},
+        new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.DOUBLE}));
+    List<Object[]> afterResultRows4 = resultTable.getRows();
+
+    validateBeforeAfterQueryResults(beforeResultRows4, afterResultRows4);
+
+    query = "SELECT column1, max(column1), sum(column10) from testTable WHERE column6 = 1001 GROUP BY "
+        + "column1 ORDER BY column1";
+    brokerResponseNative = getBrokerResponse(query);
+    assertTrue(brokerResponseNative.getProcessingExceptions() == null
+        || brokerResponseNative.getProcessingExceptions().size() == 0);
+    resultTable = brokerResponseNative.getResultTable();
+    assertEquals(brokerResponseNative.getNumRowsResultSet(), 1);
+    assertEquals(brokerResponseNative.getTotalDocs(), 400_000L);
+    assertEquals(brokerResponseNative.getNumDocsScanned(), 8);
+    assertEquals(brokerResponseNative.getNumSegmentsProcessed(), 4L);
+    assertEquals(brokerResponseNative.getNumSegmentsMatched(), 4L);
+    assertEquals(brokerResponseNative.getNumEntriesScannedPostFilter(), 16L);
+    assertEquals(brokerResponseNative.getNumEntriesScannedInFilter(), 426752);
+    assertNotNull(brokerResponseNative.getProcessingExceptions());
+    assertEquals(brokerResponseNative.getProcessingExceptions().size(), 0);
+    assertEquals(resultTable.getDataSchema(),
+        new DataSchema(new String[]{"column1", "max(column1)", "sum(column10)"}, new DataSchema.ColumnDataType[]{
+            DataSchema.ColumnDataType.INT, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE
+        }));
+    List<Object[]> afterResultRows5 = resultTable.getRows();
+
+    validateBeforeAfterQueryResults(beforeResultRows5, afterResultRows5);
   }
 
   @Test
   public void testRangeIndexAfterReload()
       throws Exception {
-    String query = "select count(*) from testTable where column10 > 674022574";
+    String query = "select count(*) from testTable where column10 > 674022574 and column9 < 674022574";
     BrokerResponseNative brokerResponseNative = getBrokerResponse(query);
     assertTrue(brokerResponseNative.getProcessingExceptions() == null
         || brokerResponseNative.getProcessingExceptions().size() == 0);
     ResultTable resultTable1 = brokerResponseNative.getResultTable();
     assertEquals(brokerResponseNative.getNumRowsResultSet(), 1);
     assertEquals(brokerResponseNative.getTotalDocs(), 400_000L);
-    assertEquals(brokerResponseNative.getNumDocsScanned(), 103280L);
+    assertEquals(brokerResponseNative.getNumDocsScanned(), 40224L);
```

Comment:
So rangeIndex will be rewritten to be raw value based right ? I wonder why should `numDocsScanned` / `numEntriesScannedInFilter` change ?

Reply:
I actually changed the query by adding another WHERE clause.